include/nbl/asset/utils/CPolygonGeometryManipulator.h

```diff
@@ -17,15 +18,31 @@ namespace nbl::asset
 class NBL_API2 CPolygonGeometryManipulator
 {
 	public:
-		//vertex data needed for CSmoothNormalGenerator
-		struct SSNGVertexData
-		{
-			uint32_t index;									     //offset of the vertex into index buffer
-			uint32_t hash;											       //
-			float wage;												         //angle wage of the vertex
-			hlsl::float32_t3 position;							   //position of the vertex in 3D space
-			hlsl::float32_t3 parentTriangleFaceNormal; //
-		};
+
+    struct SSNGVertexData
+    {
+      uint64_t index;									     //offset of the vertex into index buffer
```

Comment:
oof my bad, we only support inices up to 32bit :facepalm:

https://github.com/Devsh-Graphics-Programming/Nabla/blob/aa5d6cb90b383e68b52c6abe69c6efe96281c7dd/include/nbl/asset/IPolygonGeometry.h#L58-64

https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/asset/IPolygonGeometry.h#L215-228